packages/realm/src/schema/validate.ts

```diff
@@ -164,7 +164,10 @@ export function validatePropertySchema(
       assert.string(property, `'${propertyName}.property' on '${objectName}'`);
     }
     if (indexed !== undefined) {
-      assert.boolean(indexed, `'${propertyName}.indexed' on '${objectName}'`);
+      assert(
+        typeof indexed === "boolean" || indexed === "fulltext",
+        `"indexed" needs to be either a boolean or "fulltext"`,
```

Comment:
The change of error message should have caused [this validation test](https://github.com/realm/realm-js/blob/d9df13ff87a4c26e8e2eac188927ac4873ebc37f/packages/realm/src/tests/schema-validation.test.ts#L233C5-L240) to fail. You can run from the root `npm test --workspace realm`. Even with this change, the test should fail but the test need to be updated by changing the last part of the expected error message. (Can make a small PR against this branch to add minor things.)
```suggestion
        `Expected '${propertyName}.indexed' on '${objectName}' to be a boolean or 'fulltext'.`,
```